src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -37,4 +56,286 @@ public void testDatabaseMetaDataWrapper() throws SQLException {
         }
     }
 
+    /**
+     * Testing if driver version is matching with manifest file or not. Will be useful while releasing preview / RTW release.
+     * 
+     * //TODO: Test for capability 1.7 for JDK 1.7 and 1.8 for 1.8 //Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))" //String
+     * capability = attributes.getValue("Require-Capability");
+     * 
+     * @throws SQLServerException
+     *             Our Wrapped Exception
+     * @throws SQLException
+     *             SQL Exception
+     * @throws IOException
+     *             IOExcption
+     */
+    @Test
+    public void testDriverVersion() throws SQLServerException, SQLException, IOException {
+        String manifestFile = Utils.getCurrentClassPath() + "META-INF/MANIFEST.MF";
+        manifestFile = manifestFile.replace("test-classes", "classes");
+
+        File f = new File(manifestFile);
+
+        assumeTrue(f.exists(), "Manifest file is not exist on classpath so ignoring test");
+
+        InputStream in = new BufferedInputStream(new FileInputStream(f));
+        Manifest manifest = new Manifest(in);
+        Attributes attributes = manifest.getMainAttributes();
+        String buildVersion = attributes.getValue("Bundle-Version");
+
+        DatabaseMetaData dbmData = connection.getMetaData();
+
+        String driverVersion = dbmData.getDriverVersion();
+
+        boolean isSnapshot = buildVersion.contains("SNAPSHOT");
+
+        // Removing all dots & chars easy for comparing.
+        driverVersion = driverVersion.replaceAll("[^0-9]", "");
+        buildVersion = buildVersion.replaceAll("[^0-9]", "");
+
+        // Not comparing last build number. We will compare only major.minor.patch
+        driverVersion = driverVersion.substring(0, 3);
+        buildVersion = buildVersion.substring(0, 3);
+
+        int intBuildVersion = Integer.valueOf(buildVersion);
+        int intDriverVersion = Integer.valueOf(driverVersion);
+
+        if (isSnapshot) {
+            assertTrue(intDriverVersion < intBuildVersion, "In case of SNAPSHOT version build version should be always greater than BuildVersion");
```

Comment:
Wouldn't it be better to assert that (intDriverVersion == (intBuildVersion - 1))?